### DISCUSSION TITLE
Ability to disable the UI?

### DISCUSSION THREAD

**Discussion by rauschp** [pocketbase/pocketbase#2605] [Q&A] [answered]
I know this may be kind of a weird request, but Pocketbase is freaking _perfect_ for my project I've been working on.  I need something that's easily distributable and can comfortably run in a single docker container.  That said - one thing that I've wanted is to actually not serve the Pocketbase UI in a production environment.  Everything will be handled via my custom UI. Logs and such will still be queryable but the user/consumer of this container could likely severely mess things up if they have access to the Pocketbase UI.  Sure, that doesn't stop them from hitting the API directly.. but that requires more of an advanced user.  

Any way to do that currently, or if it doesn't exist, would a PR be supported to implement that?

Thanks in advanced!

**Comment by ganigeorgiev** [comment, marked as answer]
With the next v0.17.0 release it will be possible to overwrite existing route definitions, so technically you'll be able to register a new `/_/` route that returns 404 in your case.

**Reply by rauschp** [reply]
@ganigeorgiev This is freaking awesome! Thanks so much! I can go ahead and start using this then.

**Reply by ganigeorgiev** [reply]
Alternatively, you can also register a global middleware that will return 404 if the route path start with `/_/`, something like:

```go
package main

import (
	"log"
	"strings"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func admin404() echo.MiddlewareFunc {
	return func(next echo.HandlerFunc) echo.HandlerFunc {
		return func(c echo.Context) error {
			if strings.HasPrefix(c.Path(), "/_/") {
				return echo.ErrNotFound
			}

			return next(c)
		}
	}
}

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.Use(admin404())
		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```
